_source/_posts/2022-09-09-supabase-vue.md

```diff
@@ -0,0 +1,941 @@
+---
+layout: blog_post
+title: ""
+author:
+by: advocate|contractor
+communities: [devops,security,mobile,.net,java,javascript,go,php,python,ruby]
+description: ""
+tags: []
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness|conversion
+---
+
+## Introduction
+
+Supabase is an open-source Firebase alternative. It provides several critical functions for modern web applications such as Postgres databases, authentication, real-time subscriptions, and blob storage. The interface is very easy to use but is very feature-rich. Supabase also provides support for javascript via a client that is available via yarn or npm. Like the web platform, the client is very easy to use and integrate into any modern web application. Best of all it is free to get started with supabase and you can scale into a premium plan on demand.
+
+In this article, you will learn how to build a wellness tracker app that will allow authenticated users to create new entries for specified days. These entries will be marked as public or private. Public entries will be visible to all visitors however private entries are only available to authenticated users.
+
+The application will be built with Vue.js using tailwinds to style the site. You will use Supabase to store the data used for the site. Finally, you will use Okta to secure the application.
+
+## Prerequisites
+
+- [Visual Studio 2022 v17.1 or later](https://code.visualstudio.com/)
+- [A Supabase Account](https://app.supabase.com/)
+- [Node.js version 15.0 or higher](https://nodejs.org/en/)
+- [Okta CLI](https://cli.okta.com)
+> [Okta](https://developer.okta.com/) has Authentication and User Management APIs that reduce development time with instant-on, scalable user infrastructure. Okta's intuitive API and expert support make it easy for developers to authenticate, manage and secure users and roles in any application.
+
+You can view the [code on github](https://github.com/nickolasfisher/OktaSupabase.git).
+
+## Create your Okta Application
+
+{% include setup/cli.md type="spa" framework="Vue"
+   loginRedirectUri="http://localhost:3000/callback" signup="false" %}
+
+Copy the output from the CLI and keep it for later.
+
+## Setup your Supabase project
+
+Go to your [Supabase account](https://app.supabase.com/) and click on **New Project**. Select your default organization (or create a new one). Name your project `OktaSupabase` and give it a strong password. Select the region closest to you and select the pricing plan that suits your needs best. Most likely this will be the `Free` plan but if you find a need to update to pro you can do that as well. Once you're done click **Create Project**.
```

Comment:
```suggestion
Go to your [Supabase account](https://app.supabase.com/) and click on **New Project**. Select your default organization (or create a new one). Name your project `OktaSupabase` and give it a strong password. Select the region closest to you and select the pricing plan that suits your needs best. Most likely this will be the `Free` plan but if you find a need to update to pro you can do that as well. Once you're done, click **Create Project**.
```